lib/ws-kernel.js

```diff
@@ -27,21 +28,21 @@ export default class WSKernel extends Kernel {
   }
 
   interrupt() {
-    return this.session.kernel.interrupt();
+    this.session.kernel.interrupt();
   }
 
   shutdown() {
-    return this.session.kernel.shutdown();
+    this.session.kernel.shutdown();
   }
 
   restart(onRestarted: ?Function) {
     const future = this.session.kernel.restart();
     future.then(() => {
-      if (onRestarted) onRestarted(this.session.kernel);
+      if (onRestarted) onRestarted();
     });
   }
 
-  _execute(code: string, callWatches: boolean, onResults: Function) {
+  execute(code: string, callWatches: boolean, onResults: ResultsCallback) {
```

Comment:
Please, no API changes to ZMQKernel (that's what I've been trying to argue all along)

Reply:
👍 Above comment has been addressed in the latest patchset